### ISSUE TITLE
searching with ctrl+f doesn't work with two words

### ISSUE THREAD

**Issue by jazztickets** [mozilla/pdf.js#9736]
Attach (recommended) or Link to PDF file here:
[dee752ed0f726d8785abf360ca783d91f96f9a2e.pdf](https://github.com/mozilla/pdf.js/files/2013876/dee752ed0f726d8785abf360ca783d91f96f9a2e.pdf)

Configuration:
- Web browser and its version: Firefox 60/Chromium 66
- Operating system and its version: Linux/Windows 7
- PDF.js version: v1.10.88 or v1.9.426 or the version built into Firefox 60

Steps to reproduce the problem:
1. Hit ctrl+f and search for "pioneer of"
2. Pioneer will be highlighted, but as soon as you type a space no results are found

pdftotext shows the correct text:

```
in nit ris hington 1D C
boerge W lacan a pioneer of butali
and an influential man aw at richfield last walk
```

It works in chrome's built-in PDF viewer, so it's not a problem with the pdf.

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):
https://newspapers.lib.utah.edu/pdfjs1.9/web/viewer.html?file=/udn_files/de/e7/dee752ed0f726d8785abf360ca783d91f96f9a2e.pdf

**Comment by AbhimanyuVashisht** [comment]
I would love to work on this, 
@timvandermeij  Please help me, where to start while working on this issue

**Comment by timvandermeij** [comment]
I would suggest to first check what we have in the text layer because that may explain why the search is not working. My guess is that the space factor is not correct (see https://github.com/mozilla/pdf.js/blob/7bb066494f3f15c396941c3532b83da6d3465c6b/src/core/evaluator.js#L1303) and this is also the cause of many other open text selection issues. However, this is not really trivial to fix and we may need to check how other PDF viewers do this.

**Comment by Snuffleupagus** [comment]
Unfortunately https://github.com/mozilla/pdf.js/issues/9736#issuecomment-390629834 won't help here, since this is a scanned file where the *every* word is positioned individually with different x/y coordinates; see e.g the beginning of the `/Contents` stream:
```
BT
1 G
1 g
1 0 0 1 52 1145.37 Tm
/F1 11 Tf
(UTAH)Tj
1 0 0 1 90 1146.36 Tm
/F1 11 Tf
(NEWS)Tj
1 0 0 1 29 1124.14 Tm
/F1 10 Tf
(the)Tj
1 0 0 1 44 1126.03 Tm
/F1 8 Tf
(oregon)Tj
1 0 0 1 71 1124.47 Tm
/F1 9 Tf
(short)Tj
1 0 0 1 92 1125.47 Tm
/F1 9 Tf
(line)Tj

...
```

**Comment by RNCTX** [comment]
Hello guys, as I'm sure you're aware other PDF rendering projects suffer from this as well.  Here's an example of a file that I have worked with on other utilities.

This is a scanned excerpt from an aircraft's autopilot service manual, originally printed in the 1970s on unknown equipment.

[CenturyIIB-origscan.pdf](https://github.com/mozilla/pdf.js/files/2032351/CenturyIIB-origscan.pdf)
[CenturyIIB-tesseract_hocr-uncleaned.pdf](https://github.com/mozilla/pdf.js/files/2032358/CenturyIIB-tesseract_hocr-uncleaned.pdf)
[CenturyIIB-tesseract_hocr-cleaned.pdf](https://github.com/mozilla/pdf.js/files/2032359/CenturyIIB-tesseract_hocr-cleaned.pdf)

The first file is the original scan without a text layer.

The second is a PDF/A that has been processed with [Tesseract (v4.0) OCR](https://github.com/tesseract-ocr/), unmodified.

The third has been de-skewed with [unpaper (v6.1)](https://github.com/Flameeyes/unpaper) and then OCR'd with the same version of Tesseract and output as a PDF/A as well.

In both the second and third cases, the 'hocr' rendering option with Tesseract was used for the OCR stage.

Here are some excerpts copied/pasted from various utilities...

hocr-unlceaned on Safari 11.1 (13605.1.33.1.4)

`The Century IIB Autopilot is an "Open Loop" system which responds only to the dynamics of the aircraft in flight, thus the only ground checks that can be accomplished are functional checks as described in this bulletin.`

hocr-uncleaned on Chrome 66.0.3359.181

`The Century IIB Autopilot is an "Open Loop" system which responds only to the dynamics of the aircraft in flight, thus the only ground checks that can be accomplished are functional checks as described in this bulletin.`

hocr-uncleaned on Adobe Acrobat Pro X

```
The Century IIB Autopilot is an "Open Loop" system which responds only to the
dynamics of the aircraft in flight, thus the only ground checks that can be
accomplished are functional checks as described in this bulletin.
```

hocr-uncleaned on pdf.js (Firefox 60.0.1)

```
The
Century
IIB
Autopilot
is
an
"Open Loop"
system
which
responds
only
to
the
dynamics
of
the
aircraft
in
flight,
thus
the
only
ground
checks
that
can
be
accomplished
are
functional
checks
as
described
in
this
bulletin.
```

hocr-cleaned on the same version of Safari above

`The Century IIB Autopilot is an "Open Loop’ system which responds only to the dynamics of the aircraft in flight, thus the only ground checks that can be accomplished are functional checks as described in this bulletin.`

hocr-cleaned on the same version of Chrome above

```
The Century IIB Autopilot is an "Open Loop’ system which responds only to the
dynamics of the aircraft in flight, thus the only ground checks that can be
accomplished are functional checks as described in this bulletin. 
```

hocr-cleaned on the same version of Adobe Acrobat Pro above

```
The Century IIB Autopilot is an "Open Loop’ system which responds only to the
dynamics of the aircraft in flight, thus the only ground checks that can be
accomplished are functional checks as described in this bulletin.
```

hocr-cleaned on the same version of pdf.js (Firefox) above

```
The 
Century 
IIB 
Autopilot 
is 
an 
"Open 
Loop’ 
system 
which 
responds 
only 
to 
the 
dynamics 
of 
the 
aircraft 
in 
flight, 
thus 
the 
only 
ground 
checks 
that 
can 
be 
accomplished 
are 
functional 
checks 
as 
described 
in 
this 
bulletin. 
```

**Comment by Snuffleupagus** [comment]
WFM, most likely fixed by PR #13257.
